base/applications/mspaint/main.cpp

```diff
@@ -337,10 +339,19 @@ _tWinMain (HINSTANCE hThisInstance, HINSTANCE hPrevInstance, LPTSTR lpszArgument
     imageArea.SendMessage(WM_SIZE, 0, 0);
 
     /* by moving the window, the things in WM_SIZE are done */
-    mainWindow.SetWindowPlacement(&(registrySettings.WindowPlacement));
+    if (registrySettings.WindowPlacement.length)
+    {
+        RECT rc = registrySettings.WindowPlacement.rcNormalPosition;
+        mainWindow.MoveWindow(rc.left, rc.top,
+                              rc.right - rc.left, rc.bottom - rc.top,
+                              TRUE);
+    }
 
     /* Make the window visible on the screen */
-    ShowWindow (hwnd, nFunsterStil);
+    if (registrySettings.WindowPlacement.showCmd == SW_MAXIMIZE)
```

Comment:
I know for a fact, that this code (with all those ifs etc.) might be simplified.
For example:
- intializing to defaults the position: https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l361
(and later, getting them from registry);
- next, creating the main window already at its correct position:
https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l3041
- then just calling ShowWindow with the correct parameter:
https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=de892d5bc715d68af757a36173e9f99e7927115f#l3261

Reply:
Fixed.